child-processes/index.md

```diff
@@ -1,8 +1,189 @@
 ---
-layout: default
+layout: post
 title: Child Processes
+url: child-processes
+author: ian
 ---
 
-Coming soon.
+<<<<<<< HEAD
+_Prerequisites: [events](../events), [streams](../buffers-and-streams)_
 
-In the meantime, please see https://nodejs.org/api/child_process.html
+The typical operating system has different processes running in the background, and each process is being managed by a single-core of our CPU and will run a series of calculations each time it is being ticked. To take full advantage of our CPU using a single process, we would need a number of processes that is at least equal to the number of cores in our CPU. In addition, each process might be responsible for running a series of calculations of different logic, which will give the end user a better control over the CPU’s behavior.
+
+The `child_process` module provides the ability to spawn child processes and interact with other programs. Whilst single-threaded, non-blocking performance in Node.js is great for a single process, we can use `child_process` to handle the increasing workload in our applications in multiple threads.
+
+Using multiple processes allows a Node application to scale. Node is designed for building distributed applications with many nodes, hence it's name Node.
+
+Pipes for `stdin`, `stdout`, and `stderr` are established between the parent Node process and the spawned subprocess. The behaviour matches that of pipes in the shell.
+
+The examples in this article are all Linux-based. On Windows, you will need to switch the commands I use with their Windows alternatives or use a Linux-like shell.
+
+# `exec` vs `spawn`
+
+There are two main approaches to running child processesL `exec` and `spawn`.
+
+| `exec`                                                     | `spawn`                                                     |
+|------------------------------------------------------------|-------------------------------------------------------------|
+| spawns a shell in which the command is executed            | Does not spawn a shell                                      |
+| buffers the data (waits until the process finishes and transfers all the data ) | Streams the data returned by the child process (data flow is constant) |
+| maximum data transfer 200kb (by default                    | has no data transfer size limit                             |
+
+Typically, `spawn` is more suitable for long-running process with large outputs. spawn streams input/output with child process. `exec` buffered output in a small (by default 200K) buffer.
+
+## A Simple `exec`
+
+In this example, we will call the `ls` command to listwith the optional param `-l` to show a long list of details. 
+<div class="repl-code">
+
+```javascript
+
+const { exec } = require('child_process');
+
+exec('ls -l', (err, stdout, stderr) => {
+  console.log(stdout);
+});
+
+```
+</div>
+
```

Comment:
I'd maybe add a quick line saying what exec does here - may seem obvious but I think it helps to reinforce the point.

Reply:
Just pushed